src/resnet.jl

```diff
@@ -193,105 +200,31 @@ backbone(m::ResNet) = m.layers[1]
 classifier(m::ResNet) = m.layers[2]
 
 """
-    ResNet18(; pretrain = false, nclasses = 1000)
+    ResNet(depth = 50; pretrain = false, nclasses = 1000)
    
-Create a ResNet-18 model
+Create a ResNet model with a specified depth
 ([reference](https://arxiv.org/abs/1512.03385v1)).
-See also [`Metalhead.ResNet`](#).
+See also [`Metalhead.resnet`](#).
 
 # Arguments
+- `depth`: depth of the ResNet model. Options include (18, 34, 50, 101, 152).
 - `nclasses`: the number of output classes
 
 !!! warning
-    `ResNet18` does not currently support pretrained weights.
+    Only `ResNet(50)` currently supports pretrained weights.
 """
-function ResNet18(; pretrain = false, nclasses = 1000)
-  model = ResNet(resnet_config[:resnet18]...; block = basicblock, nclasses = nclasses)
+function ResNet(depth::Int = 50; pretrain = false, nclasses = 1000)
+    @assert depth in (18, 34, 50, 101, 152) "`depth` must be one of (18, 34, 50, 101, 152)"
```

Comment:
@darsnack This lists acceptable depth values. Anything we need to clarify here? Happy to hear your thoughts on what would be a better message.

Reply:
Ah, I'm sorry I missed this! Looks good enough for me.

Reply:
Is it possible to reuse `keys(resnet_config)` for this or is that not guaranteed to be ordered?

Reply:
`sort(keys(resnet_config))` I guess

Reply:
Done.